tokio/src/macros/cfg.rs

```diff
@@ -215,22 +215,41 @@ macro_rules! cfg_macros {
     }
 }
 
+// When running tests in loom, mocks are used instead of metrics.
 macro_rules! cfg_metrics {
     ($($item:item)*) => {
         $(
-            // For now, metrics is only disabled in loom tests.
-            // When stabilized, it might have a dedicated feature flag.
+            #[cfg(not(loom))]
+            $item
```

Comment:
To what extent can we get rid of this? It's okay if we can't easily do so, but if we can, that would simplify the code.

Reply:
Interesting, I have no idea. I had figured that a bunch of extra atomics was making loom run more slowly or something? I can see what happens if we _don't_ do that. @carllerche do you know why metrics are disabled under loom?

Reply:
I don't remember. If we don't want metrics to impact loom (which we probably don't) we can just use `std` atomics unconditionally.

Reply:
so the reason (?) it was using the loom types is that the loom types work even when AtomicU64 doesn't exist on the platform 🤔 (fallback to Mutex)

I guess the best solution would be to make an AtomicCounter type that fell back to U32 when U64 support wasn't available?

I don't think we'd want to stabilize & take a lock for all the metrics on platforms without AtomicU64 support 🤔

Maybe we should keep this as-is in this CR and follow up to remove the loom-gating